powertools-parameters/src/test/java/software/amazon/lambda/powertools/parameters/DynamoDbProviderTest.java

```diff
@@ -0,0 +1,167 @@
+        Mockito.when(client.getItem(getItemValueCaptor.capture())).thenReturn(GetItemResponse.builder()
+                .item(null)
+                .build());
+
+        // Act
+        String value = provider.getValue("key");
+
+        // Assert
+        assertThat(value).isEqualTo(null);
+    }
+
+    @Test
+    public void getValueWithMalformedRowThrows() {
+        // Arrange
+        String key = "Key1";
+        HashMap<String, AttributeValue> responseData = new HashMap<String, AttributeValue>();
+        responseData.put("id", AttributeValue.fromS(key));
+        responseData.put("not-val", AttributeValue.fromS("something"));
+        Mockito.when(client.getItem(getItemValueCaptor.capture())).thenReturn(GetItemResponse.builder()
+                .item(responseData)
+                .build());
+        // Act
+        Assertions.assertThrows(NullPointerException.class, () -> {
+            String value = provider.getValue(key);
+        });
+    }
+
+
+    @Test
+    public void getValues() {
+
+        // Arrange
+        String key = "Key1";
+        String subkey1 = "Subkey1";
+        String val1 = "Val1";
+        String subkey2 = "Subkey2";
+        String val2 = "Val2";
+        HashMap<String, AttributeValue> item1 = new HashMap<String, AttributeValue>();
+        item1.put("id", AttributeValue.fromS(key));
+        item1.put("sk", AttributeValue.fromS(subkey1));
+        item1.put("val", AttributeValue.fromS(val1));
+        HashMap<String, AttributeValue> item2 = new HashMap<String, AttributeValue>();
+        item2.put("id", AttributeValue.fromS(key));
+        item2.put("sk", AttributeValue.fromS(subkey2));
+        item2.put("val", AttributeValue.fromS(val2));
+        QueryResponse response = QueryResponse.builder()
+                .items(item1, item2)
+                .build();
+        Mockito.when(client.query(queryRequestCaptor.capture())).thenReturn(response);
+
+        // Act
+        Map<String, String> values = provider.getMultipleValues(key);
+
+        // Assert
+        assertThat(values.size()).isEqualTo(2);
+        assertThat(values.get(subkey1)).isEqualTo(val1);
+        assertThat(values.get(subkey2)).isEqualTo(val2);
+        assertThat(queryRequestCaptor.getValue().tableName()).isEqualTo(tableName);
+        assertThat(queryRequestCaptor.getValue().keyConditionExpression()).isEqualTo("id = :v_id");
+        assertThat(queryRequestCaptor.getValue().expressionAttributeValues().get(":v_id").s()).isEqualTo(key);
+    }
+
+    @Test
+    public void getValuesWithoutResultsReturnsNull() {
+        // Arrange
+        Mockito.when(client.query(queryRequestCaptor.capture())).thenReturn(
+                QueryResponse.builder().items().build());
+
+        // Act
+        Map<String, String> values = provider.getMultipleValues(UUID.randomUUID().toString());
+
+        // Assert
+        assertThat(values.size()).isEqualTo(0);
+    }
+
+    @Test
+    public void getValuesWithMalformedRowThrows() {
```

Comment:
It's not documented on the base interface what happens if you ask for something and the data that comes back is either 1/ not there or 2/ doesn't match our schema. I am asserting that for 1/ we return null and 2/ we throw.

Reply:
agree on that, but throw something else than nullpointer